scripts/jest/jest-cli.js

```diff
@@ -0,0 +1,268 @@
+      default: false,
+    },
+    build: {
+      alias: 'b',
+      describe: 'Run tests on builds.',
+      requiresArg: false,
+      type: 'boolean',
+      default: false,
+    },
+    persistent: {
+      alias: 'n',
+      describe: 'Run with persistence.',
+      requiresArg: false,
+      type: 'boolean',
+      default: false,
+    },
+    ci: {
+      describe: 'Run tests in CI',
+      requiresArg: false,
+      type: 'boolean',
+      default: false,
+    },
+  }).argv;
+
+function logError(message) {
+  console.error(chalk.red(`\n${message}`));
+}
+function validateOptions() {
+  let success = true;
+  if (argv.variant && argv.project !== 'www') {
+    logError(
+      'Variant is only supported for the www config. Update these options to continue.'
+    );
+    success = false;
+  }
+
+  if (argv.prod && argv.project === 'devtools') {
+    logError('DevTools do not support --prod. Remove this option to continue.');
+    success = false;
+  }
+
+  if (argv.env === 'production' && argv.project === 'devtools') {
+    logError(
+      'DevTools do not support --env=production. Remove this option to continue.'
+    );
+    success = false;
+  }
+
+  if (argv.build && argv.persistent) {
+    logError(
+      'Persistence is not supported for build targets. Update these options to continue.'
+    );
+    success = false;
+  }
+
+  if (argv.project !== 'oss' && argv.persistent) {
+    logError(
+      'Persistence only supported for oss configs. Update these options to continue.'
+    );
+    success = false;
+  }
+
+  if (argv.build && argv.project !== 'oss' && argv.project !== 'devtools') {
+    logError(
+      'Build targets are only supported for oss and devtools configs. Update these options to continue.'
+    );
+    success = false;
+  }
+
+  if (!argv.build && argv.project === 'devtools') {
+    logError(
+      'Source target is not supported devtools. Remove this option to continue.'
+    );
+    success = false;
+  }
+
+  if (argv.env && argv.env !== 'production' && argv.prod) {
+    logError(
+      'Build type does not match --prod. Update these options to continue.'
+    );
+    success = false;
+  }
+
+  if (argv.env && argv.env !== 'development' && argv.dev) {
+    logError(
+      'Build type does not match --dev. Update these options to continue.'
+    );
+    success = false;
+  }
+
+  if (argv.prod && argv.dev) {
+    logError(
+      'Cannot supply both --prod and --dev. Remove one of these options to continue.'
+    );
+    success = false;
+  }
+
+  if (!success) {
+    console.log(''); // Extra newline.
+    process.exit(1);
+  }
+}
+
+function getCommandArgs() {
+  // Add the correct Jest config.
+  const args = ['./scripts/jest/jest.js', '--config'];
+  if (argv.build) {
+    args.push(buildConfig);
+  } else if (argv.persistent) {
+    args.push(persistentConfig);
+  } else {
+    args.push(configMap[argv.project]);
+  }
+
+  if (argv.build) {
+    // TODO: We could build this if it hasn't been built yet.
```

Comment:
minor nit: We could check the contents of the `build` folder and give you a better warning if they're empty (or maybe even if they're older than some recent timestamp?)

Reply:
Done!